single_cell_analysis/scripts/pf.eval.kmeans.R

```diff
@@ -0,0 +1,137 @@
+# Qiwen Hu 2018
+# This script is used to evaluate the performance of dimension reduction algorithms
+# based on different types of metrics
+# Usage: Rscript pf.eval.R
+#
+# Output:
+#  performance evaluation tables and heatmap figs 
+
+library(NMI)
+library(mclust)
+library(dplyr)
+library(coRanking)
+source(file.path("util","util.R"))
+source(file.path("util", "vis.R"))
+
+# k-means based
+kmean_pf_metric <- function(input.dir, option = "real"){
+  # This function computes the k-means based performance metrics for different approaches
+  # (tybalt, tsne, umap, ZIFA and PCA) for all simulated/real single cell datasets
+  # Args:
+  #   input.dir - input directory contains feature files and celltype informations for each dataset
+  #   option - simulated/real
+  # Returns:
+  #   list contains NMI, ARI and average silhouette score
+  
+  feature.files <- list.files(input.dir)
+  
+  # list files contain celltype information
+  cellinfo.files <- feature.files[grep(paste(".*.celltype.txt", sep = ""), feature.files)]
+  
+  # tybalt depth 1
+  tybalt_d1_files <- feature.files[grep("depth1_.*tybalt_tsne", feature.files)]
+  # tybalt depth 2
+  tybalt_d2_files <- feature.files[grep("depth2_.*tybalt_tsne", feature.files)]
+  # tabalt depth 3
+  tybalt_d3_files <- feature.files[grep("depth3_.*tybalt_tsne", feature.files)]
+  tsne_files <- feature.files[grep("depth1_rnaseq_tsne", feature.files)]
+  umap_files <- feature.files[grep("umap", feature.files)]
+  zifa_files <- feature.files[grep("ZIFA", feature.files)]
+  pca_files <- feature.files[grep("pca", feature.files)]
+  
+  nmi_score <- list()
+  ari_score <- list()
+  silhouette_score <- list()
+  
+  for(i in 1:length(tybalt_d1_files)){
+    filelist <- c(tybalt_d1_files[i], tybalt_d2_files[i], tybalt_d3_files[i], tsne_files[i],
+                  umap_files[i], zifa_files[i], pca_files[i])
+    # read files
+    datalist = lapply(file.path(input.dir, filelist), function(x) read.table(x, header=T)) 
+    cellinfo <- read.table(file.path(input.dir, cellinfo.files[i]), sep = "\t", header = TRUE)
+    
+    # extract dataset information
+    #dataset <- get_param(tybalt_d1_files[i], option = option)
```

Comment:
remove?

Reply:
yea, I am not suing this function. I deleted this line.